src/eval/array.rs

```diff
@@ -271,6 +272,18 @@ impl Array {
         Ok(result)
     }
 
+    /// Zips the array with another array. If the two arrays are of unequal length, it will
+    /// only zip up until the last element of the smaller array and the rest will be ignored.
+    pub fn zip(&self, other: Array) -> Array {
+        let mut zipped_result = Array::new();
+        for (first, second) in zip(self.iter(), other.iter()) {
+            let pair =
+                Value::Array(Array::from_vec(eco_vec![first.clone(), second.clone()]));
+            zipped_result.push(pair);
+        }
+        zipped_result
```

Comment:
this would best be written in iterator style. `Array` implements `FromIterator`. we also don't need to import `zip` since `self.iter().zip(other.iter())` works just as well.

Reply:
Better now? 😄